wiki/docs/developer-resources/functional-spec.md

```diff
@@ -0,0 +1,218 @@
+
+![Access control](functional-spec-images/access-control.png)
+
+### Panel behavior
+1. By default the `default view` is shown on the satellite's display.
+1. The user can activate other views by uttering access sentences. The view is shown on the satellite's device that the user is using.
+1. After `timeout` second the satellite in use switches back to the `default view`.
+1. Access sentences can be used while on any view, even while music is playing. If at all possible we would want to detect intents from the configured `assistant` and subscribe to handle those, so we do not need any automations.
+1. Something about link icons goes here, not sure yet what they are and how they fit in.
+
+## Development plan and priorities
+|Workitem #|Workitem|Priority|Pre requisites|Exit criteria|
+|---|---|---|---|---|
+|1|Integration and entity registration|0|None|User can select, load VA integration and add one or more satellites.|
+|2|Panel|0|#1|Once VA integration is loaded, a panel is registered. Panel shows all built-in views and exhibits behavior described in [Panel behavior](#panel-behavior).|
+|3|Setting page|0|#2|Setting page can be accessed from all views in panel|
+|4|Tabstrip for settings page|0|#3||Multiple tabs can be registered on the tabstrip and can be switched|
+|5|Matrix for settings pages|0|#3|Multiple setting rows can be added, the global column is available and the sallites are shown as columns. Each satellite column cell has an override checkbox which is default off and the same input element as the global setting. If a override checkbox is not checked, any change to the corresponding global setting is reflected in the (disabled) input element. If a override checkbox is checked, the input element is enabled and any changes to the global setting is not reflected any more.|
+|6|General settings page|0|#3,#4,#5|Settings page applies settings to satellites as configured. Settings currently in blueprint are available|
+|7|Clock settings|0|#3,#4,#5|Setting page applies settings to clock view as configured|
+|8|Passcode entrybox|1|#3|Passcode setting is available to on general settings page and user can enter configured passcode to access settings panel|
+|9|Add view|1|#3|User can add a view and invoke the view using view access sentences. Custom views can be added by pasting YAML code|
+|10|Access control|2|#3|User can enable / disable views from access by satellites and those changes are reflected|
+
+Not in the table above:
+- other view settings, similar to #7
+
+## Open questions
+Lots of open questions, but here are a few:
+1. Should we provide a way to customize access sentences for built-in views? Should this be a setting on the [view specific settings page](#view-specific-settings?
+1. Is it indeed possible to listen to intent from `assistant` and have the integration act on it? This is a research item which we might have to discuss with HA devs.
```

Comment:
It would be good to see this integrated into core I think. I'll do some more research on the `assist_pipeline` component and see if I can get it to expose something on the event bus.

Event bus is important as it can contain more information about something than what a state change can do.